### PR TITLE
Ban Revoke structure

### LINKED ISSUE #280: Ban revoke structure change
https://github.com/FAForever/db/pull/176

### PR DESCRIPTION
**Pull request by 1-alex98**
Fixes #280

### CODE REVIEW

**Comment by 1-alex98** [comment]
The correct faf-stack version does not exist yet

File: src/main/java/com/faforever/api/data/domain/BanInfo.java [hunk 1]
```diff
@@ -55,8 +56,8 @@ public Player getPlayer() {
   }
 
   @ManyToOne
+  @UpdatePermission(expression = "Prefab.Role.None")
   @JoinColumn(name = "author_id")
-  @NotNull
```
Comment (Brutus5000) [inline src/main/java/com/faforever/api/data/domain/BanInfo.java]: Why was @NotNull removed here?

**Review by Brutus5000 — CHANGES_REQUESTED** [review]
Please add integration tests to BanInfoTest.java
They were missing before, because it did not work with Elide. Now as it should be working, we should have tests.

File: src/main/java/com/faforever/api/data/domain/BanInfo.java (hunk 1 above)
Comment (1-alex98) [reply src/main/java/com/faforever/api/data/domain/BanInfo.java]: It was removed because it is set now anyway so no @NotNull nedded

Comment (Brutus5000) [reply src/main/java/com/faforever/api/data/domain/BanInfo.java]: No. Implementations may change, but there is nothing more explicit than validation right on the entity. No reason to remove it. It gives additional information and enforces a validation.

**Comment by Brutus5000** [comment]
When I run this locally on Windows the following tests fail:
com.faforever.api.data.BanTest#canCreateBanInfoBasedOnModerationReportAsModerator
com.faforever.api.data.BanTest#canCreateBanInfoAsModerator
com.faforever.api.data.BanTest#canReadBanInfoAsModerator

Can you check?

**Comment by Brutus5000** [comment]
Looks like it was a local gradle issue with not updating java-commons

**Review by Brutus5000 — APPROVED** [review]

**Pull request merged**